pkg/storage/engine/temp_engine.go

```diff
@@ -52,7 +56,16 @@ func NewTempEngine(ctx context.Context, storeCfg base.StoreSpec) (Engine, error)
 		MaxOpenFiles:    128, // TODO(arjun): Revisit this.
 	}
 	rocksDBCache := NewRocksDBCache(0)
-	return NewRocksDB(rocksDBCfg, rocksDBCache)
+	r, err := NewRocksDB(rocksDBCfg, rocksDBCache)
+	return TempEngine{r}, err
+}
+
+// Close wraps Engine.Close and deletes on-disk files.
+func (t TempEngine) Close() {
+	t.Engine.Close()
+	if rocks, ok := t.Engine.(*RocksDB); ok {
+		_ = rocks.Destroy()
```

Comment:
We don't actually call `RocksDB.Destroy()` from anywhere else (AFAICT). It might be better to get rid of that and to call `os.RemoveAll` on the path.